### PR TITLE
refactor and cleanup organizational structure search

### PR DESCRIPTION
**Pull request by radoslavzeman**
Trying to cleanup MS GRAPH endpoints and fetching at least a little bit.

- use `useQuery` instead of `useSwr`
- search only on group, not in whole AD
- unify all users search to one endpoint, search not only on `displayName` but also in `jobTitle` and `mail`
- add links to docs
- reorganise and simplify files structure in `/backend`

2nd comit
- remove old rozkopavky posts from HP
- remove legacy api routes and fetchers (facebook, cycling graph - opendata...)
- unify fetchers (ginis, ms-graph)
- rename, cleanup, move files, group server related functions in `server` folders for each service
- upgrade `@azure/msal-node`, `xml2js` libs - no breaking change

TODOs:
- more cleanup in `/api/structure` endpoint
- structure should be prefetched
- handle AD outages by remembering latest fetched structure
- ...

### CODE REVIEW

**Review by mpinter — APPROVED** [review]

File: next/pages/api/ms-graph/search-in-structure.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      headers: {
+        Authorization: `Bearer ${accessToken}`,
+        ConsistencyLevel: 'eventual',
+      },
+    })
+
+    if (!response.ok) {
+      console.log(response)
+
+      return
```
[the remark this answers is not in the document]
Comment (radoslavzeman) [reply next/pages/api/ms-graph/search-in-structure.ts:58]: No it doesn't, I must've overlooked it

File: next/backend/ms-graph/fetchers/msGraphSearchFetcher.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { MSGraphFilteredGroupUser } from '@backend/ms-graph/types'
+import axios from 'axios'
+
+export const getMsGraphSearchQueryKey = (search: string) => ['msGraphUserSearch', search]
+
+export const msGraphSearchFetcher = async (searchQuery: string) => {
+  return axios.get<MSGraphFilteredGroupUser[]>(
+    `/api/ms-graph/search-in-structure?query=${searchQuery}`,
+  )
+}
```
[the remark this answers is not in the document]
Comment (radoslavzeman) [reply next/backend/ms-graph/fetchers/msGraphSearchFetcher.ts:10]: It doesn't hold anymore (for a log time), but I'll figure out some way how to make sure it's clear what should be called only from server.

**Review by mpinter — APPROVED** [review]
(povodne slack message, kopirujem sem a nechavam bez prekladu do eng 🙂 ) co sa obsahu ginisOfficialBoard.ts tyka - kedze by uz mal existovat aj json endpoint odkedy tamto vzniklo, idealne by bolo jednak to pridat do tejto kniznice https://github.com/bratislava/ginis-sdk dvak zacat z nej pouzivat ten json endpoint namiesto preparsovavania xml (v kniznici asi je stare xml ude, ale pravdepodobne uz nie up to date touto dobou kedze sa nepouzivalo)

**Comment by radoslavzeman** [comment]
I'm adding the task for refactor to json endpoint into the project :)

**Pull request merged**